### DISCUSSION TITLE
Using behind a kubernetes service with multiple replicas [13753]

### DISCUSSION THREAD

**Discussion by kristoferB** [eProsima/Fast-DDS#2458] [Q&A]
Hi,

We are running a kubernetes cluster for ROS2 and are both communicating internally in the cluster and to external ROS nodes. This works great using the DS!

However, i would like to set up a kubernetes service for the DS with multiple DS-replicas behind the service. However, from what I understand each client need to know the exact server prefix for each DS-server, and each server need a unique prefix, so a Kubernetes server cannot be used to “hide” the servers. Is this the case? Is there a way for a client to connect to a single IP address (of the Kubernetes service) and then be routed to one of the DS replicas?

Is there a better way to set up DS in Kubernetes? I had a look at using the backup function but i think it makes more sense to use replicas and a service.

**Comment by JLBuenoLopez** [comment]
Hi @kristoferB,

I am going to transfer the issue to the Fast DDS repository. This repository includes only infrastructure to test the Fast DDS Discovery Server feature.
